### DISCUSSION TITLE
Downgradin Https To Http Not Work (sslstrip.py)

### DISCUSSION THREAD

**Discussion by enrico8055** [mitmproxy/mitmproxy#6885] [General]
i try using mitmdump with sslstrip.py but the https seem not downgrading to http.

NB:
no error when running the script
in http web the proxy work perfectly
i hv already tried non hsts web such as zsecurity.org and still dont work

**Comment by Prinzhorn** [comment]
Nowadays most websites are on hardcoded HSTS lists (https://hstspreload.org/) practically making sslstrip impossible (HSTS was added to prevent this exact attack). Since you didn't give any details, I assume that's the problem.

**Reply by enrico8055** [reply]
i hv already tried non hsts web such as zsecurity.org and still dont work

**Reply by Prinzhorn** [reply]
"dont work" is not very helpful. Do work for me:

![image](https://github.com/mitmproxy/mitmproxy/assets/679144/a1a2d672-8ae9-49bd-b304-a93a15faab94)
